Wingstop_Recovery_Computer_V1.ino

```diff
@@ -0,0 +1,525 @@
+  
+  while(t2 < t_droge + 1000){// logs data for one second
+  
+  //update altitude at frequency (hz)
+  iterate_altitude();
+  
+  log_data();//logs data to sd card
+
+}
+
+  digitalWrite(FIRE_PIN_1, LOW); //turns off droge releay (IGN 1)
+
+
+
+
+
+// ***********************DESCENDING MODE 1************************************
+
+while(!detect_main()){// logs data for one second
+  
+  //update altitude at frequency (hz)
+  iterate_altitude();
+  
+  log_data();//logs data to sd card
+
+}
+
+  reopen_file();//saves and reopens file
+
+
+  
+
+// ***********************FIRE 2 (MAIN CHUTE)************************************
+
+  t_main = t2; //saves droge fire time
+  myFile.print(F("FIRE MAIN")); //logs event
+  digitalWrite(FIRE_PIN_2, HIGH); //turns on droge releay (IGN 1)
+  
+  while(t2 < t_main + 1000){// logs data for one second
+  
+  //update altitude at frequency (hz)
+  iterate_altitude();
+  
+  log_data();//logs data to sd card
+
+}
+
+  digitalWrite(FIRE_PIN_2, LOW); //turns off droge releay (IGN 1)
+
+
+
+
+
+
+// ***********************DESCENDING MODE 2************************************
+
+while(!detect_landing()){// logs data for one second
+  
+  //update altitude at frequency (hz)
+  iterate_altitude();
+  
+  log_data();//logs data to sd card
+
+}
+
+  myFile.print(F("LANDED")); //logs event
+  
+  close_file();
+
+
+
+
+
+// ***********************RECOVERY MODE************************************
+  
+  recovery_beeps();
+  
+
+
+
+}//end of main program
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+//******FUNCTIONS********FUNCTIONS********** FUNCTIONS **********FUNCTIONS*********FUNCTIONS**********FUNCTIONS*************
+
+//***********LED FUNCTIONS***********
+
+void turn_on_led(){ // Turns on LED
+  digitalWrite(LED_PIN, HIGH);
+}
+
+void turn_off_led(){ // Turns off LED
+  digitalWrite(LED_PIN, LOW);
+}
+
+
+//***********BUZZER FUNCTIONS***********
+
+void turn_on_buzzer(){ // Turns on LED
+  digitalWrite(BUZZER_PIN, HIGH);
+}
+
+void turn_off_buzzer(){ // Turns off LED
+  digitalWrite(BUZZER_PIN, LOW);
+}
+
+void beep_buzz(int num){ // Blinks LED and Beeps Buzzer for num times
+  for(int i = 0; i < num; i++){
+    turn_on_buzzer();
+    turn_on_led();
+    delay(300);
+    turn_off_buzzer();
+    turn_off_led();
+    delay(300);
+  }
+}
+
+void recovery_beeps(){//Continuous Slow long beeps for recovery
+  while(1){ //does not exit loop
+  turn_on_buzzer();
+  delay(1000); //beep delay
+  turn_off_buzzer();
+  delay(5000); //pause delay
+  }
+}
+
+
+//***********BAROMETER FUNCTIONS***********
+
+void read_barometer(){//takes barometer reading
+  while (! bmp.performReading()) {//performs barometer reading
+    //could not perform reading
+  }
+}
+
+float read_altitude(){//reads and returns barometer altitude reading
+  while (! bmp.performReading()) {
+    //performs until it gets a valid reading
+  }
+  return bmp.readAltitude(init_pressure); // returns current altitude reading
```

Comment:
We need to investigate whether this library function accounts for noisy readings from the barometer sensor.